Added iPhone X layout support (iOS 9+)

RomanTysiachnik opened the pull request:
## PR with necessary updates for iPhone X support. 

One of the new features iOS is adding Safe Areas, that should help developers with a layout on devices with irregular screens like iPhone X. 

First of all that you have to mention before applying this pull-request is that the deployment target for the project has to be at least **iOS 9.0** to make these changes work.

Though these fixes supposed to be directed only on iPhone X, it still has some UI changes that should be mentioned, like **Camera button** and **Live label**. If we launch our controls on any device with **iOS 11.0+**, they will look normal, because Safe Area is also covered other devices, adding Safe Area on the top of the screen, that provides necessary space for the status bar. But if we launch our controls on any iOS **lower than 11.0**, our two controls on the top will have the beginning of their constraint from the top of the screen, instead of the bottom margin of the status bar. 

I will leave some examples of the layout on different devices in the comments below. 

I'm looking forward to your suggestions for any changes or if you have any other ideas to implement these updates.

[JIRA Issue](https://jira.ops.aol.com/browse/OMSDK-722)

Balavor commented:
![screen shot 2017-11-20 at 13 43 10](https://user-images.githubusercontent.com/31652265/33035774-0963e61a-ce35-11e7-8966-21c150c327eb.png)
![screen shot 2017-11-20 at 13 42 07](https://user-images.githubusercontent.com/31652265/33035843-3939b32e-ce35-11e7-8a98-d5ccf6097686.png)
![screen shot 2017-11-20 at 13 47 38](https://user-images.githubusercontent.com/31652265/33035864-47993138-ce35-11e7-91a7-26cd06a492a0.png)
![screen shot 2017-11-20 at 19 23 28](https://user-images.githubusercontent.com/31652265/33035805-1afc1cb2-ce35-11e7-851d-fe37fd0cc681.png)

AndriiMoskvin commented on PlayerControls/sources/DefaultControlsViewController.swift:
```diff
@@ -92,6 +92,15 @@ public final class DefaultControlsViewController: ContentControlsViewController
     var task: URLSessionDataTask?
     
+    var shouldHideHomeIndicator = true
+    //Method that hides or shows iPhone X home indicator based on controlsView hidden status.
```
Comments usually applied through `///`

AndriiMoskvin commented:
If we connect our constraints to the bottom of status bar - will this still work only on iOS 9+?

RomanTysiachnik commented:
Added new updates to this pull-request:

- dropped unnecessary files
- rebased branch on last merge commit and got rid of merge conflicts
- detached shadowView from controls to make its constraints relevant to the View instead of its Safe Area.

invalid404 commented on PlayerControls/sources/DefaultControlsViewController.swift:
```diff
@@ -111,7 +110,7 @@ public final class DefaultControlsViewController: ContentControlsViewController
+        if #available(iOS 11.0, *) {
             if controlsView.isHidden {
                 shouldHideHomeIndicator = true
                 setNeedsUpdateOfHomeIndicatorAutoHidden()
```
Code duplication

RomanTysiachnik replied:
@invalid404 probably a bug after rebase, because this PR is frozen.

invalid404 commented on PlayerControls/sources/DefaultControlsViewController.swift:
```diff
@@ -111,7 +110,7 @@ public final class DefaultControlsViewController: ContentControlsViewController
         controlsView.isHidden = uiProps.controlsViewHidden
-        if #available(iOS 11.0, *)  {
+        if #available(iOS 11.0, *) {
             if controlsView.isHidden {
```
Could you rewrite without if statement?

RomanTysiachnik commented:
@aol-public/mobile-sdk-team 

Updates for 01.03.2018:
- [x] Added safe areas for Player Controls in a current version of xib;
- [x] Changed animation constraints for the safe area;
- [x] Added fix for sidebar bottom constraint;
- [ ]  Snapshot tests safe area support (Snappy Shrimp framework bug).

MarkGerl commented:
@AndriiMoskvin can we get this updated before your next iOS drop?  I really like this, and it's great!

AndriiMoskvin commented:
@MarkGerl We are going to merge this - so 100%.

RomanTysiachnik commented:
Added a new PR #102 from the current repo branch instead of the fork one.
